### PR TITLE
UX Refurbishing

### PR DESCRIPTION
**Pull request by chrismgeorge**
## Updates to UX
* File Manager
  * Snippet update in modal, mv-snippet pinned to top
* Editor
  * Poster, throws a toast if the initial camera isn't set when poster is generated
* Camera
  * Fov is never set
  * Minimum zoom is settable (sets min fov and min radius)
  * Removed sliders for radius and fov independently.
  * Set initial camera is aware if it's bounds are out of range (radius, yaw, pitch) and its background will change if it is out of range.
* Pinned at top of each tab is a relevant section (i.e. snippet or material you are editing)

### CODE REVIEW

File: packages/space-opera/editor/index.html [hunk 1]
```diff
@@ -46,6 +46,23 @@
   <script async src='https://www.google-analytics.com/analytics.js'></script>
 </head>
 <style>
+  :root {
```
Comment (chrismgeorge) [inline packages/space-opera/editor/index.html:49]: using global colors now, to adjust light/dark theme

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts [hunk 2]
```diff
@@ -18,50 +18,190 @@
...
+
+  open() {
+    this.isOpen = true;
+  }
+
+  close() {
+    this.isOpen = false;
+  }
+
+  render() {
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:124]: https://screenshot.googleplex.com/BiXpFSvhjBBSpQe.png

File: packages/space-opera/src/components/poster_controls/poster_controls.ts [hunk 3]
```diff
@@ -24,80 +24,99 @@ import {safeDownloadCallback} from '@google/model-viewer-editing-adapter/lib/uti
...
-    reduxStore.dispatch(
-        dispatchSaveCameraOrbit(currentOrbit, currentFieldOfViewDeg));
+
+    // if we've already set the initial camera, use it
+    if (this.cameraSnippet.orbit !== undefined) {
+      const initialOrbit = this.cameraSnippet.orbit;
+      modelViewer.cameraOrbit = getOrbitString(initialOrbit);
+      modelViewer.jumpCameraToGoal();
+      modelViewer.requestUpdate();
+      requestAnimationFrame(async () => {
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/poster_controls/poster_controls.ts:91]: is this the correct way to use requestAnimationFrame?

File: packages/space-opera/src/components/model_viewer_snippet/model_viewer_snippet.ts [hunk 4]
```diff
@@ -89,129 +93,65 @@ export class ExportPanel extends ConnectedLitElement {
...
-    width: 95%;
-  }
-        `;
-  }
-
-  @query('textarea#mv-input') private readonly textArea!: HTMLInputElement;
-
-  @internalProperty() errors: string[] = [];
-
-  async handleSubmitSnippet(event: Event) {
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/model_viewer_snippet/model_viewer_snippet.ts:133]: moved to open_button

File: packages/space-opera/src/styles.css.ts [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {css, CSSResult} from 'lit-element';
```
Comment (chrismgeorge) [inline packages/space-opera/src/styles.css.ts:18]: most css for components here

File: packages/space-opera/src/test/camera_settings/components/radius_limits_test.ts [hunk 6]
```diff
@@ -1,75 +0,0 @@
...
-
-import '../../../components/camera_settings/components/radius_limits.js';
-
-import {DEFAULT_MIN_RADIUS, RadiusLimits} from '../../../components/camera_settings/components/radius_limits.js';
-import {dispatchInitialCameraState, dispatchRadiusLimits, getCamera} from '../../../components/camera_settings/reducer.js';
-import {getModelViewer} from '../../../components/model_viewer_preview/model_viewer.js';
-import {ModelViewerPreview} from '../../../components/model_viewer_preview/model_viewer_preview.js';
-import {reduxStore} from '../../../space_opera_base.js';
-
-xdescribe('radius limits editor test', () => {
```
Comment (chrismgeorge) [inline packages/space-opera/src/test/camera_settings/components/radius_limits_test.ts:27]: this feature was removed

File: packages/space-opera/src/test/camera_settings/components/fov_limits_test.ts [hunk 7]
```diff
@@ -1,72 +0,0 @@
...
-
-import '../../../components/camera_settings/components/fov_limits.js';
-
-import {DEFAULT_MAX_FOV, FovLimits} from '../../../components/camera_settings/components/fov_limits.js';
-import {dispatchFovLimits, getCamera} from '../../../components/camera_settings/reducer.js';
-import {getModelViewer} from '../../../components/model_viewer_preview/model_viewer.js';
-import {ModelViewerPreview} from '../../../components/model_viewer_preview/model_viewer_preview.js';
-import {reduxStore} from '../../../space_opera_base.js';
-
-xdescribe('fov limits editor test', () => {
```
Comment (chrismgeorge) [inline packages/space-opera/src/test/camera_settings/components/fov_limits_test.ts:27]: this feature was removed (was testing sliders for fov)

File: packages/space-opera/src/components/home/types.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  dropdownBackground: string;
+  numberInputBackground: string;
+}
+
+interface ThemeInterface {
+  dark: CSSAttributes;
+  light: CSSAttributes;
+}
+
+export const THEMES: ThemeInterface = {
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/home/types.ts:89]: light and dark theme defined here

**Comment by chrismgeorge** [comment]
Obviously, I'd like to hear about the UI and UX. :)

File: packages/space-opera/src/components/camera_settings/camera_settings.ts [hunk 9]
```diff
@@ -184,6 +211,19 @@ export class CameraSettings extends ConnectedLitElement {
     reduxStore.dispatch(dispatchRadiusLimits(radiusLimits));
   }
 
+  undoInitialCamera() {
```
Comment (elalish) [inline packages/space-opera/src/components/camera_settings/camera_settings.ts:214]: undo is a bit vague; do you mean reset?

File: packages/space-opera/src/components/camera_settings/components/limits_base.ts [hunk 10]
```diff
@@ -23,15 +23,17 @@ import '../../shared/checkbox/checkbox.js';
...
 import {SliderWithInputElement} from '../../shared/slider_with_input/slider_with_input.js';
 import {Limits} from '../types.js';
 
 import {styles} from './limits_base.css.js';
 
+
 /** Abstract component that can be extended for editing scalar limits. */
 export abstract class LimitsBase extends ConnectedLitElement {
-  static styles = styles;
+  static styles = [styles, snackbarStyles];
```
Comment (elalish) [inline packages/space-opera/src/components/camera_settings/components/limits_base.ts:36]: The snackbar strikes again!

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts [hunk 11]
```diff
@@ -157,7 +157,8 @@ export class OpenModal extends LitElement {
             <textarea id="mv-input" rows=10>${exampleLoadableSnippet}</textarea>
             ${this.errors.map(error => html`<div>${error}</div>`)}
           </div>
-          <div>
+          <div class="mv-note">Edit the top snippet. Then update the bottom snippet, which sets the &lt;model-viewer&gt paramters in the editor.</div>
+          <div class="update-button">
```
Comment (elalish) [inline packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: I think this two-snippet thing is unnecessary: The done button should do what the update button does. Might be nice to have a cancel button too. Actually, I think this modal thing should just be for snippet input, and put a top-level button in for GLB import; that'll be more symmetric with the GLB and ZIP export buttons below.

File: packages/space-opera/src/components/poster_controls/poster_controls.ts [hunk 12]
```diff
@@ -35,60 +37,86 @@ import {getCameraState} from '../model_viewer_preview/model_viewer_preview.js';
...
-          <mwc-button unelevated class="PosterButton"
-            @click="${this.onDownloadPoster}">Download</mwc-button>
-          <mwc-button unelevated class="PosterButton"
-            @click="${this.onDisplayPoster}">Display Poster</mwc-button>
-          <mwc-button unelevated class="PosterButton"
-            @click="${this.onDeletePoster}">Delete Poster</mwc-button>` :
+        <mwc-button unelevated class="PosterButton"
+          @click="${this.onDownloadPoster}">Download Poster</mwc-button>
+        <mwc-button unelevated class="PosterButton"
+          @click="${this.onDisplayPoster}">Redisplay Poster</mwc-button>
```
Comment (elalish) [inline packages/space-opera/src/components/poster_controls/poster_controls.ts:67]: I think Display is better.

File: packages/space-opera/src/components/poster_controls/poster_controls.ts (hunk 3 above)
Comment (elalish) [reply packages/space-opera/src/components/poster_controls/poster_controls.ts:91]: LGTM, though I think you're missing a `;`. I think `requestUpdate()` is superfluous (since RAF invokes it), but you can test that pretty easily.

File: packages/space-opera/src/components/poster_controls/poster_controls.ts [hunk 13]
```diff
@@ -35,60 +37,86 @@ import {getCameraState} from '../model_viewer_preview/model_viewer_preview.js';
...
-    // TODO, fix such that we aren't modifying modelViewer in here. i.e. add
-    // settings to modelViewerPreview
-    const src = modelViewer.src;
     // Normally we can just use dispatchSetReveal, but the value has to be
     // changed immediately before reload.
     modelViewer.reveal = 'interaction';
-    // Force reload the model
-    modelViewer.src = '';
-    modelViewer.src = src;
+    modelViewer.showPoster()
```
Comment (elalish) [inline packages/space-opera/src/components/poster_controls/poster_controls.ts:119]: 💯

File: packages/space-opera/src/test/camera_settings/camera_settings_test.ts [hunk 14]
```diff
@@ -81,9 +80,7 @@ describe('camera constraints test', () => {
     const yawInput = cameraSettings.cameraOrbitEditor!.yawInput!;
     expect(yawInput).toBeDefined();
     expect(yawInput).not.toBeNull();
-    yawInput.setValue(42);
-    const stateOrbit = getCamera(reduxStore.getState()).orbit;
-    expect(stateOrbit!.thetaDeg).toBeCloseTo(42);
+    // Now has a dependency on model-viewer.
```
Comment (elalish) [inline packages/space-opera/src/test/camera_settings/camera_settings_test.ts:83]: What is the dependency?

File: packages/space-opera/editor/index.html (hunk 1 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:49]: Global colors is great, but I may have to veto the light theme. First, the two themes are not consistent: lights uses lines for borders while dark uses contrasting backgrounds. Second, having two themes doubles the maintenance for adding features (or if we get a11y feedback), even long after you've moved on from the project.

File: packages/space-opera/editor/index.html [hunk 15]
```diff
@@ -86,50 +103,57 @@
...
     <div class="editor-body-root">
       <div class="mvContainer">
         <model-viewer-preview id="editing_adapter">
         </model-viewer-preview>
       </div>
       <me-tabs>
-        <me-tabbed-panel icon="save">
-          <model-viewer-snippet></model-viewer-snippet>
+        <me-tabbed-panel icon="home">
+          <home-container></home-container>
```
Comment (elalish) [inline packages/space-opera/editor/index.html:132]: I don't feel like this home screen adds anything; it's out of context and unclickable and only serves to hide the way a user gets started, which is by opening a 3D model. I think the tab descriptors would better serve as tooltips on the tabs themselves.

File: packages/space-opera/editor/index.html [hunk 16]
```diff
@@ -86,50 +103,57 @@
...
-        <me-tabbed-panel icon="color_lens">
-          <me-materials-panel></me-materials-panel>
-        </me-tabbed-panel>
         <me-tabbed-panel icon="photo_camera">
+          <me-export-panel header="true"></me-export-panel>
           <me-camera-settings></me-camera-settings>
         </me-tabbed-panel>
+        <me-tabbed-panel icon="color_lens">
+          <me-materials-panel></me-materials-panel>
+        </me-tabbed-panel>
```
Comment (elalish) [inline packages/space-opera/editor/index.html:151]: The inspector panel is slightly wider than the others and causes the preview to resize.

File: packages/space-opera/editor/index.html (hunk 1 above)
Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:49]: Here are my thoughts on that. I actually think having two themes would increase the long-term productivity of anyone working on this editor moving forward. When introducing a feature, it is extremely easy to get into the habit of copy, pasting a random hex value throughout the code, instead of using a global color. Having a light theme is a reminder that you must use global colors, which I believe makes it easier to keep a unified color scheme throughout the app. In a sense, it forces uniformity, and eases updates in the future. Yes, just having a global color scheme does this as well. But I believe the Light theme forces the global scheme to be used correctly. 

If you still don't think we should have it, it's easy enough to remove, your call. (And I can change the Light Theme cards to have a background contrast instead of border, easy enough as well.)

File: packages/space-opera/editor/index.html [hunk 17]
```diff
@@ -86,50 +103,57 @@
...
           <me-ibl-selector></me-ibl-selector>
           <me-animation-controls></me-animation-controls>
           <me-hotspot-panel></me-hotspot-panel>
           <me-poster-controls></me-poster-controls>
         </me-tabbed-panel>
-        <me-tabbed-panel icon="color_lens">
-          <me-materials-panel></me-materials-panel>
-        </me-tabbed-panel>
         <me-tabbed-panel icon="photo_camera">
+          <me-export-panel header="true"></me-export-panel>
```
Comment (elalish) [inline packages/space-opera/editor/index.html:146]: UX feedback on camera panel: 
image

File: packages/space-opera/src/components/poster_controls/poster_controls.ts (hunk 3 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/poster_controls/poster_controls.ts:91]: done

File: packages/space-opera/editor/index.html [hunk 18]
```diff
@@ -86,50 +103,57 @@
...
         </model-viewer-preview>
       </div>
       <me-tabs>
-        <me-tabbed-panel icon="save">
-          <model-viewer-snippet></model-viewer-snippet>
+        <me-tabbed-panel icon="home">
+          <home-container></home-container>
+        </me-tabbed-panel>
+        <me-tabbed-panel icon="import_export">
+          <me-export-panel></me-export-panel>
```
Comment (elalish) [inline packages/space-opera/editor/index.html:135]: Let's put the snippet at the top of the export panel to be consistent with the two following panels.

File: packages/space-opera/src/components/camera_settings/camera_settings.ts (hunk 9 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/camera_settings/camera_settings.ts:214]: yep, done

File: packages/space-opera/editor/index.html (hunk 17 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:146]: Also there's some kind of async bug because I'm seeing a lot of "Waiting for camera target" when a model is loaded and functional.

File: packages/space-opera/src/components/camera_settings/components/limits_base.ts (hunk 10 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/camera_settings/components/limits_base.ts:36]: lol: https://stackoverflow.com/questions/34432339/android-snackbar-vs-toast-usage-and-difference
tldr; Toasts are snackbars without actionable content

Comment (chrismgeorge) [reply packages/space-opera/src/components/camera_settings/components/limits_base.ts:36]: so technically yes, this should be named toast

File: packages/space-opera/src/components/poster_controls/poster_controls.ts (hunk 12 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/poster_controls/poster_controls.ts:67]: done

File: packages/space-opera/editor/index.html [hunk 19]
```diff
@@ -86,50 +103,57 @@
...
         </me-tabbed-panel>
-        <me-tabbed-panel icon="color_lens">
-          <me-materials-panel></me-materials-panel>
-        </me-tabbed-panel>
         <me-tabbed-panel icon="photo_camera">
+          <me-export-panel header="true"></me-export-panel>
           <me-camera-settings></me-camera-settings>
         </me-tabbed-panel>
+        <me-tabbed-panel icon="color_lens">
+          <me-materials-panel></me-materials-panel>
```
Comment (elalish) [inline packages/space-opera/editor/index.html:150]: Materials tab UX feedback:
image

Comment (elalish) [reply packages/space-opera/editor/index.html:150]: Mostly, let's get consistent ordering and minimize the vertical space used by stacking things more horizontally.

File: packages/space-opera/src/test/camera_settings/camera_settings_test.ts (hunk 14 above)
Comment (chrismgeorge) [reply packages/space-opera/src/test/camera_settings/camera_settings_test.ts:83]: This calls the function: onCameraOrbitEditorChange(), which requires knowledge of a model-viewer element to be in active. I still haven't figured out the async issues revolving around why we can't test the model-viewer on the full test suite.

This test was previously just testing the effect that moving the slider had on the redux state, nothing to do with model-viewer.

**Review by elalish — CHANGES_REQUESTED** [review]

File: packages/space-opera/src/test/camera_settings/camera_settings_test.ts (hunk 14 above)
Comment (elalish) [reply packages/space-opera/src/test/camera_settings/camera_settings_test.ts:83]: Well, if you're going to put a comment, try to put enough info so someone other than you can know what you mean. Is this a TODO? Is there a footgun to be aware of?

File: packages/space-opera/src/components/camera_settings/components/limits_base.ts (hunk 10 above)
Comment (elalish) [reply packages/space-opera/src/components/camera_settings/components/limits_base.ts:36]: lol, that's good to know. I'd never have guessed from their names! I guess this is how we know I'm not a UX one...

File: packages/space-opera/editor/index.html (hunk 1 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:49]: Yeah, agreed on the global colors, but I'm happy to enforce that through code review. Maintaining the Light Theme is still too much maintenance (and review).

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts (hunk 11 above)
Comment (elalish) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: Also, it's a bit confusing that loading a "default model" is done by updating the text snippet. We already had a feature request to have a dropdown to open a few provided models like the old interactive example did. It would be great if we could put that feature in so that the snippet didn't need to provide that functionality.

File: packages/space-opera/src/components/poster_controls/poster_controls.ts (hunk 12 above)
Comment (elalish) [reply packages/space-opera/src/components/poster_controls/poster_controls.ts:67]: Also, when you display the poster, you need to reset the user's camera to the initial camera so that when they interact, the transition is seamless.

File: packages/space-opera/editor/index.html (hunk 18 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:135]: Also, let's avoid the word "Upload" in all of our UX; it implies we're taking their model to our server, which we are not. "Open" is better, possibly "Import".

File: packages/space-opera/editor/index.html (hunk 16 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:151]: Doesn't seem to repro for the astronaut; I'm guessing if the JSON lines are too long it wraps wrong or something?

File: packages/space-opera/src/components/poster_controls/poster_controls.ts (hunk 12 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/poster_controls/poster_controls.ts:67]: gotcha, was gonna ask about that

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts (hunk 11 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: yep, done

Comment (chrismgeorge) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: new modal: https://screenshot.googleplex.com/6pXzkVAPbg3hPdR.png

Comment (chrismgeorge) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: new file manager: https://screenshot.googleplex.com/6qx8SgxxmQBoSGq.png

File: packages/space-opera/src/test/camera_settings/camera_settings_test.ts (hunk 14 above)
Comment (chrismgeorge) [reply packages/space-opera/src/test/camera_settings/camera_settings_test.ts:83]: updated,

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts (hunk 11 above)
Comment (elalish) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:161]: Those are looking good! Is it possible to have the buttons on their own instead of inside a named box? I feel like the box and name aren't adding much but extra space.

File: packages/space-opera/editor/index.html (hunk 16 above)
Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:151]: yep, fixed

File: packages/space-opera/editor/index.html (hunk 18 above)
Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:135]: done

File: packages/space-opera/editor/index.html (hunk 19 above)
Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:150]: The texture and color picker being horizontal with the header and revert button just doesn't look good. I know it adds additional vertical space, but the alternative is a really janky looking row of things of multiple sizes. If I make everything the same size, then it gets even more awkward with text too big or buttons too small. I think this is something we can live with for the time being.

File: packages/space-opera/editor/index.html (hunk 17 above)
Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:146]: The camera target is a pre-existing bug that will take a day or two to track down. I don't know if it's worth adding to this PR. It may very well go away when I factor out the initial camera (which I was planning on doing in a separate PR)

File: packages/space-opera/editor/index.html (hunk 19 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:150]: Can you just revert it to how it was originally? I this case I feel like the original editor had a cleaner layout.

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts (hunk 11 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts]: It would still need some kind of descriptor next to it anyway. I think given what we are asking it to do: Be a quickly recognizable feature, I think it's okay.

File: packages/space-opera/editor/index.html [hunk 20]
```diff
@@ -109,6 +109,16 @@
     overflow-y: auto;
     display: block;
   }
+  .privacy {
```
Comment (chrismgeorge) [inline packages/space-opera/editor/index.html:112]: privacy moved to bottom of first tab

Comment (chrismgeorge) [reply packages/space-opera/editor/index.html:112]: new front design: https://screenshot.googleplex.com/ZB7mzHY2B2SA6PW.png

File: packages/space-opera/src/components/model_viewer_snippet/model_viewer_snippet.ts [hunk 21]
```diff
@@ -121,16 +128,20 @@ export class ExportPanel extends ConnectedLitElement {
     <snippet-viewer id="snippet-header" .renderedSnippet=${snippet}
       .renderedStyle=${this.hotspots.length > 0 ? hotspotStyles.cssText : ``}>
     </snippet-viewer>
+    <mwc-button unelevated
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/model_viewer_snippet/model_viewer_snippet.ts:131]: moved edit snippet button to be under export snippet on first tab (seen in image from prev comment)

File: packages/space-opera/src/components/shared/slider_with_input/slider_with_input.css.ts [hunk 22]
```diff
@@ -39,6 +39,8 @@ export const styles: CSSResult = css`:host {
 paper-slider {
   --paper-slider-knob-color: var(--google-grey-300);
   --paper-slider-active-color: var(--google-grey-300);
+  --paper-slider-knob-start-color: var(--google-grey-300);
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/shared/slider_with_input/slider_with_input.css.ts:42]: fixes slides to be same color

File: packages/space-opera/src/components/shared/slider_with_input/slider_with_input.ts [hunk 23]
```diff
@@ -75,6 +75,7 @@ export class SliderWithInputElement extends LitElement {
         min="${this.min}"
         max="${this.max}"
         step="${this.step}"
+        secondary-progress="-180"
```
Comment (chrismgeorge) [inline packages/space-opera/src/components/shared/slider_with_input/slider_with_input.ts:78]: remove blue from slider for negative numbers

**Comment by mounirlamouri** [comment]
self tagging to [hopefully] get updates @mounirlamouri

**Comment by chrismgeorge** [comment]
@elalish is this an error I need to worry about: https://github.com/google/model-viewer/runs/1421387671?check_suite_focus=true#step:9:766 (Keeps popping up in model-viewer here)

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts [hunk 24]
```diff
@@ -114,21 +130,35 @@ export class OpenModal extends LitElement {
...
+      editedConfig.src = `Change this to your GLB URL`;
+    }
+    if (editedConfig.environmentImage &&
+        isObjectUrl(editedConfig.environmentImage)) {
+      // Uploaded env image
+      editedConfig.environmentImage = `Change this to your HDR URL`;
+    }
+    const exampleLoadableSnippet =
+        renderModelViewer(editedConfig, {}, undefined);
+    ////
```
Comment (elalish) [inline packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:149]: What's up with all the slashes?

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts [hunk 25]
```diff
@@ -114,21 +130,35 @@ export class OpenModal extends LitElement {
...
 
     return html`
 <paper-dialog id="file-modal" modal ?opened=${this.isOpen}>
   <div class="FileModalContainer">
     <div class="FileModalHeader">
-      <div>Edit &lt;model-viewer&gt; Snippet</div>
+      <div>Edit Snippet</div>
     </div>
-    <div style="font-size: 14px; font-weight: 500; margin-top: 10px; color: white">Edit&#47; Paste &lt;model-viewer&gt Snippet</div>
+    <div style="font-size: 14px; font-weight: 500; margin-top: 10px; color: white">Edit or paste a &lt;model-viewer&gt snippet</div>
```
Comment (elalish) [inline packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:157]: 👍

File: packages/space-opera/src/components/ibl_selector/ibl_selector.ts [hunk 26]
```diff
@@ -146,7 +146,7 @@ export class IblSelector extends ConnectedLitElement {
                 environmentImage.uri}>${environmentImage.name}</paper-item>`)}
           </me-dropdown>
           <mwc-button class="UploadButton" id="uploadButton" unelevated
-        icon="cloud_upload" @click="${this.openFileModal}">Upload</mwc-button>
+        icon="cloud_upload" @click="${this.openFileModal}">HDR</mwc-button>
```
Comment (elalish) [inline packages/space-opera/src/components/ibl_selector/ibl_selector.ts:149]: Let's make this icon match the import GLB icon.

**Review by elalish — COMMENTED** [review]
Yeah, I'm seeing those test failures too; it's not your fault. I fixed some flakes, but I think there are some remaining (they seem to have gotten worse with Edge 86 for some reason). I'll see if I can make a quick PR to squash them. I'll have some more detailed visual feedback soon, but it's looking great overall!

File: packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts (hunk 24 above)
Comment (chrismgeorge) [reply packages/space-opera/src/components/model_viewer_snippet/components/open_button.ts:149]: just trying to bind a comment, will remove them

File: packages/space-opera/editor/index.html (hunk 17 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:150]: Screen Shot 2020-11-18 at 5 29 33 PM

File: packages/space-opera/editor/index.html (hunk 18 above)
Comment (elalish) [reply packages/space-opera/editor/index.html:133]: Screen Shot 2020-11-18 at 5 37 32 PM

File: packages/space-opera/editor/index.html [hunk 27]
```diff
@@ -89,47 +96,68 @@
...
           <me-poster-controls></me-poster-controls>
         </me-tabbed-panel>
-        <me-tabbed-panel icon="color_lens">
-          <me-materials-panel></me-materials-panel>
-        </me-tabbed-panel>
         <me-tabbed-panel icon="photo_camera">
+          <me-export-panel header="true"></me-export-panel>
           <me-camera-settings></me-camera-settings>
         </me-tabbed-panel>
+        <me-tabbed-panel icon="color_lens">
```
Comment (elalish) [inline packages/space-opera/editor/index.html:153]: Can we move the metallicRoughness texture below its factors to match the others?

**Review by elalish — CHANGES_REQUESTED** [review]
A few minor details.

**Comment by chrismgeorge** [comment]
Same-size glb button, privacy doesn't overlap: https://screenshot.googleplex.com/A6Hn8fzvZa8PL3a.png
draggable inputs new size: https://screenshot.googleplex.com/7qg7j97iy8d9kBU.png

**Comment by chrismgeorge** [comment]
added remaining hdr images and made layout of dropdown + button same as on import/export page: https://screenshot.googleplex.com/7XLjwocgEMYTQXk.png

**Review by elalish — APPROVED** [review]
Nice work! This is a significant improvement to usability.
